creme/tree/hoeffding_tree_classifier.py

```diff
@@ -0,0 +1,340 @@
+    tie_threshold
+        Threshold below which a split will be forced to break ties.
+    leaf_prediction
+        | Prediction mechanism used at leafs.
+        | 'mc' - Majority Class
+        | 'nb' - Naive Bayes
+        | 'nba' - Naive Bayes Adaptive
+    nb_threshold
+        Number of instances a leaf should observe before allowing Naive Bayes.
+    nominal_attributes
+        List of Nominal attributes identifiers. If empty, then assume that all numeric attributes
+        should be treated as continuous.
+    **kwargs
+        Other parameters passed to river.tree.DecisionTree.
+
+    Notes
+    -----
+    A Hoeffding Tree [^1] is an incremental, anytime decision tree induction algorithm that is
+    capable of learning from massive data streams, assuming that the distribution generating
+    examples does not change over time. Hoeffding trees exploit the fact that a small sample can
+    often be enough to choose an optimal splitting attribute. This idea is supported mathematically
+    by the Hoeffding bound, which quantifies the number of observations (in our case, examples)
+    needed to estimate some statistics within a prescribed precision (in our case, the goodness of
+    an attribute).
+
+    A theoretically appealing feature of Hoeffding Trees not shared by other incremental decision
+    tree learners is that it has sound guarantees of performance. Using the Hoeffding bound one
+    can show that its output is asymptotically nearly identical to that of a non-incremental
+    learner using infinitely many examples.
+
+    Implementation based on MOA [^2].
+
+    References:
+    .. [1] G. Hulten, L. Spencer, and P. Domingos. Mining time-changing data streams.
+       In KDD’01, pages 97–106, San Francisco, CA, 2001. ACM Press.
+
+    .. [2] Albert Bifet, Geoff Holmes, Richard Kirkby, Bernhard Pfahringer.
+       MOA: Massive Online Analysis; Journal of Machine Learning Research 11: 1601-1604, 2010.
+
+    Examples
+    --------
+    >>> # Imports
+    >>> from skmultiflow.data import SEAGenerator
+    >>> from skmultiflow.trees import HoeffdingTreeClassifier
+    >>>
+    >>> # Setting up a data stream
+    >>> stream = SEAGenerator(seed=1)
+    >>>
+    >>> # Setup Hoeffding Tree estimator
+    >>> ht = HoeffdingTreeClassifier()
+    >>>
+    >>> # Setup variables to control loop and track performance
+    >>> n_samples = 0
+    >>> correct_cnt = 0
+    >>> max_samples = 200
+    >>>
+    >>> # Train the estimator with the samples provided by the data stream
+    >>> while n_samples < max_samples and stream.has_more_samples():
+    >>>     X, y = stream.next_sample()
+    >>>     y_pred = ht.predict(X)
+    >>>     if y[0] == y_pred[0]:
+    >>>         correct_cnt += 1
+    >>>     ht = ht.partial_fit(X, y)
+    >>>     n_samples += 1
+    >>>
+    >>> # Display results
+    >>> print('{} samples analyzed.'.format(n_samples))
+    >>> print('Hoeffding Tree accuracy: {}'.format(correct_cnt / n_samples))
```

Comment:
Does this work? If so can you add the results to the docstring?

Reply:
Examples are WIP. They will be simplified.

Reply:
As soon as the generators are merged I'll start working on the examples. :D

Reply:
The examples are in place!